test.js

```diff
@@ -1,8 +1,31 @@
 import test from 'ava';
 import execa from 'execa';
 
+function randomName() {
+	return `asdasfgrgafadsgaf${Math.random().toString().slice(2)}`;
+}
+
 test(async t => {
-	const pkgName = `asdasfgrgafadsgaf${Math.random().toString().slice(2)}`;
-	const ret = await execa('./cli.js', [pkgName, '--color'], {cwd: __dirname});
+	const ret = await execa('./cli.js', [randomName(), '--color'], {cwd: __dirname});
 	t.regex(ret.stdout, /is available/);
 });
+
+test(async t => {
+	const pkgName = 'chalk';
+	try {
+		const ret = await execa('./cli.js', [pkgName, '--color'], {cwd: __dirname});
+		t.regex(ret.stdout, /is unavailable/);
+	} catch (err) {
+		t.ok(err);
```

Comment:
I don't think the exit code is important for the test as `execa` handles that. You only have to check if an error occurs and what message was printed.